test/xref2/shadow.t/run.t

```diff
@@ -0,0 +1,8 @@
+This test contains a carefully crafted couple of modules that cause a clash of names in the current shadowing code.
+If the shadowing code is working correctly this should not crash
+
+  $ ocamlc -c -bin-annot -I . c.mli
+  $ ocamlc -c -bin-annot -I . d.mli
+  $ odoc compile -I . c.cmti
+  $ odoc compile -I . d.cmti
```

Comment:
Should also look at the result with `odoc_print`.

Reply:
Could do! I didn't put much thought into this - it was an example provided by Luke M that was failing in a nasty way, so I was satisfied that it didn't die horribly after this PR.

Reply:
On reflection I don't think I'll do this at this point, that it doesn't crash horribly is sufficient for now.